mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageManager.java

```diff
@@ -748,6 +748,9 @@ public MessageResultIterator getMessages(MessageRange set, FetchGroup fetchGroup
 
     @Override
     public Iterator<MessageUid> search(SearchQuery query, MailboxSession mailboxSession) throws MailboxException {
+        if (query.equals(new SearchQuery(SearchQuery.all()))) {
+            return getUids(mailboxSession);
```

Comment:
The proposal from @mbaechler  to avoid duplicating logic for getting List<MessageUid>. I only partially agree...